src/plugins/table-of-contents/table-of-contents-display.js

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import { sectionDisplayProps } from '../../ui/default-prop-types.js';
+
+export default function TableOfContentsDisplay({ content }) {
+  const [nodes, setNodes] = useState([]);
+
+  const { minLevel, maxLevel } = content;
+
+  useEffect(() => {
+    const nodeList = [...window.document.querySelectorAll('[data-header="true"]')]
+      .map(element => ({
+        id: element.id,
+        text: element.textContent,
+        level: Number.parseInt(element.tagName.replace(/\D/g, ''), 10) || null
+      }))
+      .filter(node => typeof node.level === 'number' && node.level >= minLevel && node.level <= maxLevel);
+
+    setNodes(nodeList);
+  }, [minLevel, maxLevel]);
+
+  if (!nodes.length) {
+    return null;
+  }
```

Comment:
I guess we could not early return and just let the display component render whatever it found as nodes, even nothing if there's nothing.
The early return made me think that there might be a danger of using `nodes` beyond this point, but there isn't.